// Copyright 2019 Zeppelin Bend Pty Ltd
// This file is part of cimbend.
// 
// cimbend is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
// 
// cimbend is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with cimbend.  If not, see <https://www.gnu.org/licenses/>.


syntax = "proto3";
option java_multiple_files = true;
option java_package = "com.zepben.protobuf.cim.iec61970.base.wires";
package zepben.cim.iec61970.base.wires;

import "zepben/cim/iec61968/common/Location.proto";
import "zepben/cim/iec61970/base/core/Terminal.proto";
import "zepben/cim/iec61970/base/diagramlayout/DiagramObject.proto";
import "zepben/cim/iec61970/base/meas/Control.proto";
import "zepben/cim/iec61970/base/meas/Measurement.proto";

/**
 * A linear shunt compensator has banks or sections with equal admittance values.
 */
message LinearShuntCompensator {

    /**
     * Master resource identifier issued by a model authority. The mRID is unique within an exchange context. Global uniqueness
     * is easily achieved by using a UUID,  as specified in RFC 4122, for the mRID. The use of UUID is strongly recommended.
     * <p>
     * For CIMXML data files in RDF syntax conforming to IEC 61970-552 Edition 1, the mRID is mapped to rdf:ID or rdf:about attributes
     * that identify CIM object elements.
     */
    string mRID = 1;

    /**
     * The name is any free human readable and possibly non unique text naming the object.
     */
    string name = 2;

    /**
     * If true, the equipment is in service.
     */
    bool inService = 3;

    /**
     * If true, the equipment is normally in service.
     */
    bool normallyInService = 4;

    /**
     * Base voltage of this conducting equipment.  Use only when there is no voltage level container used and only one base
     * voltage applies.  For example, not used for transformers.
     */
    string baseVoltageMRID = 5;

    /**
     * Conducting equipment have terminals that may be connected to other conducting equipment terminals via connectivity nodes
     * or topological nodes.
     */
    repeated iec61970.base.core.Terminal terminals = 6;

    /**
     * Zero sequence shunt (charging) susceptance per section
     */
    double b0PerSection = 7;

    /**
     * Positive sequence shunt (charging) susceptance per section
     */
    double bPerSection = 8;

    /**
     * Zero sequence shunt (charging) conductance per section
     */
    double g0PerSection = 9;

    /**
     * Positive sequence shunt (charging) conductance per section
     */
    double gPerSection = 10;

    /**
     * Shunt compensator sections in use.
     * Starting value for steady state solution. Non integer values are allowed to support continuous variables.
     * The reasons for continuous value are to support study cases where no discrete shunt compensators has yet been
     * designed, a solutions where a narrow voltage band force the sections to oscillate or accommodate for a continuous
     * solution as input.
     */
    double sections = 11;

    /**
     * Specifies the regulation status of the equipment.  True is regulating, false is not regulating.
     */
    bool controlEnabled = 12;

    /**
     * The diagram objects that are associated with the domain object.
     * <p>
     * The diagram objects are returned in a map sorted by the diagram they live in.
     */
    repeated iec61970.base.diagramlayout.DiagramObject diagramObjects = 16;

    /**
     * Datasheet information for this power system resource.
     */
    string assetInfoMRID = 17;

    /**
     * Location of this power system resource.
     */
    iec61968.common.Location location = 18;

    /**
     * The controller outputs used to actually govern a regulating device, e.g. the magnetization of a synchronous machine or
     * capacitor bank breaker actuator.
     */
    iec61970.base.meas.Control control = 19;

    /**
     * The measurements associated with this power system resource.
     */
    iec61970.base.meas.Measurement measurement = 20;

}
